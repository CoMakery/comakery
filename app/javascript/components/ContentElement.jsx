import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Element = styled.div`
  display: flex;
  flex-direction: column;
  padding-right: 40px;
`

const Title = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 3px;
`

const Children = styled.div`
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
`

class ContentElement extends React.Component {
  render() {
    return (
      <React.Fragment>
        <Element>
          <Title>
            {this.props.title}
          </Title>
          <Children>
            {this.props.children}
          </Children>
        </Element>
      </React.Fragment>
    )
  }
}

ContentElement.propTypes = {
  title: PropTypes.string.isRequired
}
ContentElement.defaultProps = {
  title: ''
}
export default ContentElement
