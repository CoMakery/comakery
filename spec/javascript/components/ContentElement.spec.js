import React from 'react'
import { mount } from 'enzyme'
import ContentElement from 'components/ContentElement'

describe('ContentElement', () => {
  it('renders component correctly', () => {
    const wrapper = mount(<ContentElement />)

    expect(wrapper).toMatchSnapshot()
  })
})
