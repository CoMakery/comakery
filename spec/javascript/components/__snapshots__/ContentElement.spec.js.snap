// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentElement renders component correctly 1`] = `
<ContentElement
  title=""
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gPWqlF",
            "rules": Array [
              "
  display: flex;
  flex-direction: column;
  padding-right: 40px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa gPWqlF"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "lastClassName": "ekDaoc",
                  "rules": Array [
                    "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: bold;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #3a3a3a;
  text-transform: uppercase;
  margin-bottom: 3px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bwzfXH ekDaoc"
            />
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": true,
                  "lastClassName": "vZQMM",
                  "rules": Array [
                    "
  font-family: Montserrat;
  font-size: 10px;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  letter-spacing: normal;
  color: #4a4a4a;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat vZQMM"
            />
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</ContentElement>
`;
